### PR TITLE
Admin interface

### PR DESCRIPTION
**Pull request by julsbreakdown**

### CODE REVIEW

File: admin/c2cgeoportal_admin/views/interfaces.py [hunk 1]
```diff
@@ -17,3 +25,59 @@
...
+
+
+@view_defaults(match_param='table=interfaces')
+class InterfacesViews(AbstractViews):
+    _list_fields = [
+        _list_field('name'),
+        _list_field('description'),
+        _list_field(
+            'layers',
+            renderer=lambda layers_group:', '.join([p.name or ''
```
Comment (arnaud-morvan) [inline admin/c2cgeoportal_admin/views/interfaces.py]: layers_group => interface

Comment (arnaud-morvan) [reply admin/c2cgeoportal_admin/views/interfaces.py]: p => l

File: admin/c2cgeoportal_admin/views/interfaces.py [hunk 2]
```diff
@@ -17,3 +25,59 @@
...
+                                                    for p in layers_group.layers])),
+        _list_field(
+            'theme',
+            renderer=lambda themes_group: ', '.join(['{}'.format(t.name, t.name) or ''
+                                                     for t in themes_group.theme]))
```
Comment (arnaud-morvan) [inline admin/c2cgeoportal_admin/views/interfaces.py]: themes_group => interface
themes_group.theme => interface.themes

File: admin/acceptance_tests/interface_test.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                'rowCount': 10
+            },
+            status=200
+        ).json['rows'][0]
+
+        first_interface = interface_test_data['interfaces'][0]
+
+        assert first_interface.id == int(first_row['_id_'])
+        assert first_interface.name == first_row['name']
+        assert first_interface.description == first_row['description']
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/interface_test.py]: I would like to see tests on complex columns (columns with custom renderer like themes, layers)

File: admin/acceptance_tests/interface_test.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    def test_grid_search(self, test_app):
+        # search on metadatas key and value parts
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/interface_test.py]: comment does not match with code

File: admin/acceptance_tests/interface_test.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    def test_edit(self, test_app, interface_test_data, dbsession):
+        interface = interface_test_data['interfaces'][0]
+        resp = test_app.get('/interfaces/{}'.format(interface.id), status=200)
+        form = resp.form
+
+        assert str(interface.id) == self.get_first_field_named(form, 'id').value
+        assert 'hidden' == self.get_first_field_named(form, 'id').attrs['type']
+        assert interface.name == self.get_first_field_named(form, 'name').value
+        assert str(interface.description or '') == self.get_first_field_named(form, 'description').value
```
Comment (arnaud-morvan) [inline admin/acceptance_tests/interface_test.py]: You should submit your form and check resulting sqlalchemy instance

File: admin/acceptance_tests/interface_test.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    from c2cgeoportal_commons.models.main import Interface, Theme, OGCServer, LayerWMS
+
+    dbsession.begin_nested()
+
+    themes = []
+    for i in range(0, 5):
+        theme = Theme(name='theme_{}'.format(i),
+                      ordering=1)
+        dbsession.add(theme)
+        themes.append(theme)
```
Comment (cmangeat) [inline admin/acceptance_tests/interface_test.py]: Wonder if dbsession.add(theme) if usefull as dbsession.add(interface) may do the job (not sure)

Comment (arnaud-morvan) [reply admin/acceptance_tests/interface_test.py]: Yes it should, when you add an instance state to the session, related instances are added too.

File: admin/acceptance_tests/interface_test.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        first_interface = interface_test_data['interfaces'][0]
+
+        assert first_interface.id == int(first_row['_id_'])
+        assert first_interface.name == first_row['name']
+        assert first_interface.description == first_row['description']
+        assert len(first_interface.layers) == 2
+        assert first_interface.layers[1].name == 'layer_wms_4'
+        assert len(first_interface.themes) == 2
+        assert first_interface.themes[1].name == 'theme_0'
+
```
Comment (cmangeat) [inline admin/acceptance_tests/interface_test.py]: assertion has to operate on 'first_row['layers']' and test data ?

File: admin/acceptance_tests/layers_wms_test.py [hunk 8]
```diff
@@ -217,7 +217,7 @@ def test_edit(self, test_app, layer_wms_test_data, dbsession):
             'public': True,
             'geo_table': 'new_geo_table',
             'exclude_properties': 'property1,property2',
-            'ogc_server_id': '2',
+            'ogc_server_id': '6',
```
Comment (cmangeat) [inline admin/acceptance_tests/layers_wms_test.py]: 'str(layer_wms_test_data['servers'][2].id) ' should be a better fix ?

File: admin/acceptance_tests/layers_wms_test.py [hunk 9]
```diff
@@ -254,7 +254,7 @@ def test_submit_new(self, dbsession, test_app):
                 'public': True,
                 'geo_table': 'new_geo_table',
                 'exclude_properties': 'property1,property2',
-                'ogc_server_id': '2',
+                'ogc_server_id': '6',
```
Comment (cmangeat) [inline admin/acceptance_tests/layers_wms_test.py]: itou

**Comment by arnaud-morvan** [comment]
@julsbreakdown : It is preferable to rebase on master intead of merging master into your branch.

**Pull request merged**